client/http/http.go

```diff
@@ -335,8 +335,14 @@ func (h *httpClient) Get(ctx context.Context, round uint64) (client.Result, erro
 			resC <- httpGetResponse{nil, fmt.Errorf("decoding response: %w", err)}
 			return
 		}
-		if len(randResp.Sig) == 0 || len(randResp.PreviousSignature) == 0 {
-			resC <- httpGetResponse{nil, fmt.Errorf("insufficient response")}
+
+		if len(randResp.Sig) == 0 {
+			resC <- httpGetResponse{nil, fmt.Errorf("insufficient response - signature is not present")}
+			return
+		}
+
+		if !h.chainInfo.Scheme.DecouplePrevSig && len(randResp.PreviousSignature) == 0 {
```

Comment:
Im really not a fan on this coupling of the beacon logic to the client that's what I want to avoid at most possible cost: now the client is required to know the specific details of each scheme. 
Why are we checking this here ? It should fail nonetheless when we verify the randomness so why checking this here ?

Reply:
@nikkolasg I can assure you those checks were there before Zondax start working on drand. If you think they are not suitable there, I can remove them. Indeed I will do it.